### PR TITLE
Add add student alert to dashboard

### PR DESCRIPTION
**Pull request by bluwave**
refs: MBL-13963
affects: Parent
release note: none

Test plan:
1. make sure you can add(pair) a student on both the dashboard and the manage children screen

### CODE REVIEW

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by inst-danger** [comment]
## Latest Installable Builds

(will redirect to this page if no builds exist yet)
student: [Link](https://rebrand.ly/yeehwzh) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2Fyeehwzh)
teacher: [Link](https://rebrand.ly/63jkubs) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2F63jkubs)
parent: [Link](https://rebrand.ly/091fe3h) | [QR](https://qr.rebrandly.com/v1/qrcode?shortUrl=rebrand%2Ely%2F091fe3h)

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1021639&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|1
Medium|17
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---
High|1

**Comment by inst-danger** [comment]
<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Fails</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:no_entry_sign:</td>
      <td>

  Build failed, skipping coverage check
  </td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>

  This pull request will not generate a release note.
  </td>
    </tr>
  </tbody>
</table>


#### Affected Apps: Parent

[MBL-13963](https://instructure.atlassian.net/browse/MBL-13963)

<details><summary>:x: Swift lint</summary>


    ⚠️  /Users/vagrant/git/Parent/Parent/Settings/AddStudentController.swift:27:1: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    /Users/vagrant/git/Parent/Parent/Settings/AddStudentController.swift:32:1: warning: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    /Users/vagrant/git/Parent/Parent/Settings/AddStudentController.swift:48:1: warning: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    ⚠️  /Users/vagrant/git/Parent/Parent/Settings/SettingsViewController.swift:51:1: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    Linting 'DiscussionTopic.swift' (21/112)
    Linting 'DiscussionTopic+Network.swift' (22/112)
    ⚠️  /Users/vagrant/git/Parent/ParentUnitTests/Settings/AddStudentControllerTests.swift:32:1: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    /Users/vagrant/git/Parent/ParentUnitTests/Settings/AddStudentControllerTests.swift:36:1: warning: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    /Users/vagrant/git/Parent/ParentUnitTests/Settings/AddStudentControllerTests.swift:43:1: warning: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    ⚠️  /Users/vagrant/git/Parent/ParentUnitTests/Settings/AddStudentControllerTests.swift:47:1: Trailing Whitespace Violation: Lines should not have trailing whitespace. (trailing_whitespace)
    /Users/vagrant/git/Parent/ParentUnitTests/Settings/AddStudentControllerTests.swift:34:81: warning: Unused Closure Parameter Violation: Unused parameter "error" in a closure should be replaced with _. (unused_closure_parameter)
    /Users/vagrant/git/Parent/ParentUnitTests/Settings/AddStudentControllerTests.swift:54:81: warning: Unused Closure Parameter Violation: Unused parameter "error" in a closure should be replaced with _. (unused_closure_parameter)
    ⚠️  /Users/vagrant/git/Parent/ParentUnitTests/Settings/AddStudentControllerTests.swift:69:81: Unused Closure Parameter Violation: Unused parameter "error" in a closure should be replaced with _. (unused_closure_parameter)
    Linting 'DashboardViewControllerTests.swift' (99/112)
    Linting 'ConversationCoursesActionSheetTests.swift' (100/112)
</details>

<p align="right">
  Generated by :no_entry_sign: <a href="http://github.com/danger/danger-js/">dangerJS</a>
</p>

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1021640&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|17
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

File: Parent/Parent/Settings/AddStudentController.swift [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        }))
+        guard let vc = presentingViewController else { return }
+        env.router.show(alert, from: vc, options: .modal())
+    }
+
+    private func addPairingCode(code: String) {
+        let request = PostObserveesRequest(userID: "self", pairingCode: code)
+        env.api.makeRequest(request) { [weak self] _, _, error in
+            guard let self = self, let vc = self.presentingViewController else { return }
+            DispatchQueue.main.async {
```
Comment (vanwagonet) [inline Parent/Parent/Settings/AddStudentController.swift]: If you use `performUIUpdate` instead, you won't need to `drainMainQueue()` in the tests.

File: Parent/Parent/Settings/AddStudentController.swift [hunk 2]
```diff
@@ -0,0 +1,63 @@
...
+    }
+
+    @objc func actionAddStudent() {
+        let title = NSLocalizedString("Add Student", comment: "")
+        let message = NSLocalizedString("Input the student pairing code provided to you.", comment: "")
+        let alert = UIAlertController(title: title, message: message, preferredStyle: .alert)
+        alert.addTextField { tf in
+            tf.placeholder = NSLocalizedString("Pairing Code", comment: "")
+        }
+        alert.addAction(AlertAction(NSLocalizedString("Cancel", comment: ""), style: .cancel, handler: { _ in }))
```
Comment (vanwagonet) [inline Parent/Parent/Settings/AddStudentController.swift]: You can omit the no-op `handler` parameter.

**Comment by instructure-cx** [comment]
Scan submitted to Checkmarx

**Comment by instructure-cx** [comment]
### Checkmarx scan completed
[Full Scan Details](https://cxmgr.inssec.info/CxWebClient/ViewerMain.aspx?scanid=1021784&projectid=17)
#### Checkmarx Scan Summary
Severity|Count
---|---
High|0
Medium|17
Low|0
Informational|0

#### Violation Summary
Severity|Count
---|---

**Pull request merged**